### ISSUE TITLE
about the training code

### ISSUE THREAD

**Issue by yxqlwl** [cszn/IRCNN#2]
Hi, sorry for interrupting. Could you please kindly provide the training code as well? Thank you~

**Comment by cszn** [comment]
See https://github.com/cszn/DnCNN/tree/master/TrainingCodes

**Comment by SuhailAliyar** [comment]
I am an image processing student. I would like implement this for MRI images. i have these doubts after going through the work
1. About the framework used for training.
2. Much details about the training details.
3. I doubts on overfitting with just 7 layers.
4. Is denoising, deblurring and super resolution integrate together on this work. Or the architecture for 3 operations are different.
5. which is the feature descriptor using here.

It will be very useful for me if you can help me.

Thanks.

**Comment by my1347** [comment]
@zuie21 
1.MatConvNet
2.Refer to DnCNN TrainingCodes
3.Using less layers is not related to overfitting I think.
4.Read the papper
5.feature descriptor? here?

are you a 医生吗

**Comment by SuhailAliyar** [comment]
We are not getting much info from the read me note.  

Regards,
Suhail Aliyar

**Comment by XSLXANDY** [comment]
Sorry, I'm confused.
Is the training codes the same as the DnCNN? But DnCNN doesn't use HQS for training.
The loss (Equation 10 in paper) is divided into 2 equations (Equation 7 and 8 in paper). Then we need to update z{k+1} by using matconvnet. I have no idea what loss should I choose when I wanna update the net. Besides, I don't know what Mathematical expression the Φ term is. 
Hope to get your help. Thanks a lot.

**Comment by cszn** [comment]
@XSLXANDY IRCNN is a model-based optimization method. It is not an end-to-end training method. IRCNN plugs the CNN denoisers into the HQS inference. So, you only need to train the denoisers.
